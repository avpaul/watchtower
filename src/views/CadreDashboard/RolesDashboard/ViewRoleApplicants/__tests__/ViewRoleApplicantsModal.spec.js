import React from 'react';
import { mount } from 'enzyme';
import ViewRoleApplicantsModal from '../viewRoleApplicantsModal';

const props = {
  loading: false,
  data: {
    applications: [
      {
        applicant: {
          id: 56,
          role: 'admin',
          sims_project_technology: 'Node/React',
          apprenticeship_technology: 'Node/Agular'
        }
      }
    ]
  },
  getARole: jest.fn(),
  roleId: 1,
  toggle: jest.fn(),
  open: true,
  title: 'Title'
};
const wrapper = mount(<ViewRoleApplicantsModal {...props} />);
describe('Role Applicants Modal', () => {
  it('should  match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should show loading', () => {
    wrapper.setProps({ loading: true });
    wrapper.instance().forceUpdate();
    const loading = wrapper.find(`[testId="loading-indicator"]`);
    expect(loading).toBeTruthy();
  });
});
