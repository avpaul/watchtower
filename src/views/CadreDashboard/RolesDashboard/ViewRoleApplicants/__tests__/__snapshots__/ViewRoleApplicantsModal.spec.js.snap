// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Role Applicants Modal should  match the snapshot 1`] = `
<ViewRoleApplicantsModal
  data={
    Object {
      "applications": Array [
        Object {
          "applicant": Object {
            "apprenticeship_technology": "Node/Agular",
            "id": 56,
            "role": "admin",
            "sims_project_technology": "Node/React",
          },
        },
      ],
    }
  }
  getARole={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  loading={false}
  open={true}
  roleId={1}
  title="Title"
  toggle={[MockFunction]}
>
  <Modal
    handleClose={[MockFunction]}
    show={true}
    showBtn={true}
    size="large"
    title="Applicants for Title"
  >
    <div
      className="modal display__modal"
    >
      <section
        className="modal__body"
      >
        <p
          className="modal__title"
        >
          Applicants for Title
        </p>
        <button
          className="modal-close"
          onClick={[MockFunction]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
          <span
            className="sr-only"
          >
            Close
          </span>
        </button>
        <div
          className="modal__body-card"
        >
          <FellowRolesCard
            key="56"
            role={
              Object {
                "apprenticeship_technology": "Node/Agular",
                "id": 56,
                "role": "admin",
                "sims_project_technology": "Node/React",
              }
            }
          >
            <div
              className="fellow-role-card"
            >
              <div
                className="fellow-role-card-top"
              >
                <div
                  className="fellow-role-card-top__image"
                >
                  <img
                    alt=""
                  />
                </div>
                <div
                  className="fellow-role-cord-top__eng-name"
                >
                  undefined undefined
                </div>
              </div>
              <div
                className="fellow-role-card-bottom"
              >
                <div
                  className="fellow-details"
                >
                  <div
                    className="cohort-details"
                  >
                    <p>
                      Cohort
                    </p>
                    <p />
                  </div>
                  <div
                    className="role-details"
                  >
                    <p>
                      Availability
                    </p>
                    <p>
                      Unavailable
                    </p>
                  </div>
                </div>
                <div
                  className="fellow-stacks"
                >
                  <div
                    className="fellow-stacks__pills"
                    key="Node"
                  >
                    Node
                  </div>
                  <div
                    className="fellow-stacks__pills"
                    key="React"
                  >
                    React
                  </div>
                  <div
                    className="fellow-stacks__pills"
                    key="Agular"
                  >
                    Agular
                  </div>
                </div>
              </div>
            </div>
          </FellowRolesCard>
        </div>
      </section>
    </div>
  </Modal>
</ViewRoleApplicantsModal>
`;
